README.md

```diff
@@ -1 +1,72 @@
-# noir_backend_using_gnark
\ No newline at end of file
+# Using gnark as a backend for Noir
+
+If you want to develop a backend for Noir, basically what you need to do is to translate ACIR (Abstract Circuit Intermediate Representation) to something that your backend could understand.
+The prerequisites for this are to have certain knowledge about ACIR and its codebase and to have certain knowledge about the backend that you are going to support.
+In this case, the backend is [gnark](https://github.com/ConsenSys/gnark). Written in Go, gnark is a fast zk-SNARK library that offers a high-level API to design circuits. The library is open-source and developed under the Apache 2.0 license 
+
+## Overview
+
+```mermaid
+flowchart
+    N[Noir] --ACIR--> BR[Backend in gnark wrapper in Rust]
+    BR[Backend in gnark wrapper written in Rust] --Raw R1CS--> B[Concrete Backend]
+    subgraph B[Concrete Backend]
+        direction TB
+        BW[WASM/FFI] --> BG[Bakend in gnark written in Go]
+    end
+```
+
+### Backend wrapper written in Rust
+
+This module is needed because a Noir's backend has to be written in Rust and we want to develop a backend using gnark which is written in Go.
+
+This Rust module is basically in charge of implementing the trait `Backend` for a given type which we've named `Gnark` which represents our backend and it is also in charge of translating ACIR into something useful. Even though gnark supports several proving systems (like Plonk and Groth16) this wrapper will translate ACIR into R1CS. In the future, more proving systems can be easily supported (we are already planning on supporting Plonk).
```

Comment:
```suggestion
This Rust module is basically in charge of implementing the trait `Backend` for a given type which we've named `Gnark` which represents our backend and of translating ACIR into something useful. Even though gnark supports several proving systems (like Plonk and Groth16) this wrapper will translate ACIR into R1CS. In the future, more proving systems can be easily supported (we are already planning on supporting Plonk).
```

If Plonk's is already supported why are we planning on supporting it at a later date? It should be explained.
Also, what do you mean when you say that it aims to "translating ACIR into something useful", why it isn't useful as it already is?

Reply:
1. A C++ Plonk implementation is already supported as a backend for Noir, but gnark's Plonk implementation isn't. We want to support several implementations.
2. What do you think about changing "translating ACIR into something useful" to "mapping ACIR into an R1CS-friendly data structure"?

Reply:
That change sounds good!